File: frontend/src/types/league.ts
import { User } from './user';

export interface League {
  id: number;
  name: string;
  description?: string | null;
  invite_code: string;
  created_by_id: number;
  created_at: string;
  member_count?: number;
  is_member?: boolean;
  is_admin?: boolean;
}

export interface LeagueMember {
  id: number;
  league_id: number;
  user_id: number;
  is_admin: boolean;
  joined_at: string;
  user_name?: string;
  user_email?: string;
}

export interface LeagueDetail extends League {
  members: LeagueMember[];
}

export interface LeagueCreate {
  name: string;
  description?: string;
}

export interface LeagueUpdate {
  name?: string;
  description?: string;
}

export interface JoinLeagueRequest {
  invite_code: string;
}

File: backend/app/schemas/league.py
from pydantic import BaseModel, ConfigDict, Field
from typing import Optional, List
from datetime import datetime


# League Member schemas
class LeagueMemberBase(BaseModel):
    user_id: int
    is_admin: bool = False


class LeagueMemberCreate(LeagueMemberBase):
    league_id: int


class LeagueMemberResponse(LeagueMemberBase):
    id: int
    league_id: int
    joined_at: datetime
    user_name: Optional[str] = None
    user_email: Optional[str] = None
    
    model_config = ConfigDict(from_attributes=True)


# League schemas
class LeagueBase(BaseModel):
    name: str
    description: Optional[str] = None
    songs_per_round: int = Field(default=1, ge=1, le=5)


class LeagueCreate(LeagueBase):
    pass


class LeagueUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    songs_per_round: Optional[int] = Field(None, ge=1, le=5)


class LeagueResponse(LeagueBase):
    id: int
    invite_code: str
    created_by_id: int
    created_at: datetime
    member_count: Optional[int] = None
    is_member: Optional[bool] = None
    is_admin: Optional[bool] = None
    
    model_config = ConfigDict(from_attributes=True)


class LeagueDetailResponse(LeagueResponse):
    members: List[LeagueMemberResponse] = []
    
    model_config = ConfigDict(from_attributes=True)


class JoinLeagueRequest(BaseModel):
    invite_code: str
